### PR TITLE
Feature/cumulus 3070

### PR DESCRIPTION
**Pull request by Jkovarik**
Full Merge PR for CUMULUS-3070

This PR contains previously approved PRs (see commit history), plus main merge updates/changes to fix merge conflicts.

### CODE REVIEW

**Comment by Jkovarik** [comment]
Awaiting merge of https://github.com/nasa/cumulus/pull/3175

**Review by Jkovarik — COMMENTED** [review]
This PR needs resolution to https://github.com/nasa/cumulus/pull/3175  to merge.

File: packages/db/tests/translate/test-granules.js [hunk 1]
```diff
@@ -853,6 +1144,7 @@ test('translateApiGranuleToPostgresGranule converts API granule to Postgres', as
     pdr_cumulus_id: pdrCumulusId,
     processing_end_date_time: new Date(apiGranule.processingEndDateTime),
     processing_start_date_time: new Date(apiGranule.processingStartDateTime),
+    // TODO -- Test this writes to PG and returns the right 'thing'
```
Comment (npauzenga) [inline packages/db/tests/translate/test-granules.js:1147]: TODONE?

**Review by npauzenga — APPROVED** [review]
This looks right to me @Jkovarik. I ran through the changes and compared to the original PRs. After the merge conflicts on the CL are resolved I think we can merge.

**Pull request merged**